src/pages/en/guides/cms/contentful.md

```diff
@@ -0,0 +1,445 @@
+
+- **Content type:** `blogPost`
+- **Title:** `Astro is amazing!`
+- **Slug:** `astro-is-amazing`
+- **Description:** `Astro is a new static site generator that is blazing fast and easy to use.`
+- **Date:** `2021-10-01`
+- **Content:** `This is my first blog post!`
+
+Feel free to add as many blog posts as you want. Now that you have some data in your Contentful space, switch to your favorite code editor to start hacking with Astro!
+
+### Creating astro components
+
+Create a new astro component `Card.astro` inside the `src/components` directory of your project. This component will be used to display your blog post cards on the homepage later on.
+
+:::tip
+Not familiar with `.astro` files? Check out [astro components](/en/core-concepts/astro-components/) for more information.
+:::
+
+```astro title="src/components/Card.astro"
+---
+interface Props {
+  title: string;
+  description: string;
+  url: string;
+  date: string;
+}
+
+const { url, title, description, date } = Astro.props;
+---
+<li>
+  <a href={`/posts/${url}/`}>
+    <h2>{title}</h2>
+  </a>
+  <time>{date}</time>
+  <p>{description}</p>
+</li>
+```
+
+Create a new interface called `blogPostFields` and add it to your `contentful.ts` file in `src/lib`. This interface will match the fields of your blog post content type in Contentful and it will be used to type your blog post entries.
+
+```ts title="src/lib/contentful.ts" ins={2,4-10}
+import contentful from "contentful";
+import type { Document } from "@contentful/rich-text-types";
+
+export interface blogPostFields {
+  title: string;
+  date: string;
+  description: string;
+  content: Document;
+  slug: string;
+}
+
+export const contentfulClient = contentful.createClient({
+  space: import.meta.env.CONTENTFUL_SPACE_ID,
+  accessToken: import.meta.env.DEV
+    ? import.meta.env.CONTENTFUL_PREVIEW_TOKEN
+    : import.meta.env.CONTENTFUL_DELIVERY_TOKEN,
+  host: import.meta.env.DEV ? "preview.contentful.com" : "cdn.contentful.com",
+});
+
+```
+
+Now that you have your `Card.astro` component and your blog entries types set up, you can start modifying the homepage `index.astro` in `src/pages` to list your blog posts.
+
+Starting with the component script section, import the `blogPostFields` interface and `contentfulClient` from `src/lib/contentful.ts`. 
+
+The `contentfulClient` will be used to fetch your blog posts from Contentful. `blogPostFields` will be passed to the `getEntries` method to type the blog posts response.
+
+```astro title="src/pages/index.astro"
+---
+import { contentfulClient } from "../lib/contentful";
+import type { blogPostFields } from "../lib/contentful";
+
+const { items } = await contentfulClient.getEntries<blogPostFields>({
+  content_type: "blogPost",
+});
+```
+
+Now that you have your blog entries fetched, you can map your reponse to get only the properties that you need. 
```

Comment:
I think we should describe this differently: we wouldn't need to make a new array that has only the properties we need, because we only refer to the ones we need in the template.

i.e., we could've done this, with no `posts` array:
```jsx
    <ul>
      {items.map((item) => (
        <Card
          title={item.fields.title}
          description={item.fields.description}
          date={new Date(item.fields.date).toLocaleDateString()}
          url={item.fields.slug}
        />
      ))}
    </ul>
```

But this looks annoying, as we have to call `.fields` every time, and convert the date inline. So it's helpful to make a `posts` array, but not for the reason of excluding properties. Maybe:

```suggestion
Now that you have your blog entries fetched, you can map them to their fields and convert the date to a readable string:
```

Reply:
Yeah, same! I didn't want to clutter the template markup. I guess mapping gives you a nice IntelliSense 😆.